sdk/python/lib/test/test_resource.py

```diff
@@ -136,6 +136,28 @@ def check_opts(i, name, opts):
         in enumerate(depends_on_variations(dep))
     ])
 
+@pulumi.runtime.test
+def test_depends_on_typechecks_sync():
+    # https://github.com/pulumi/pulumi/issues/13917
+    try:
+        res = MockResource(name='res', opts=pulumi.ResourceOptions(depends_on=["hello"]))
+        assert False, "should of failed"
+    except TypeError as e:
+        assert str(e) == "'depends_on' was passed a value hello that was not a Resource."
```

Comment:
Do these tests need a bare except below this with `assert False` to ensure we did not get another exception type?

Reply:
Any other exception will bubble past the except and fail the test anyway